openquake/server/views.py

```diff
@@ -145,6 +148,46 @@ def get_engine_version(request):
     return HttpResponse(oqversion)
 
 
+@csrf_exempt
+@cross_domain_ajax
+@require_http_methods(['POST'])
+def validate_nrml(request):
+    """
+    Leverage oq-risklib to check if a given xml text is a valid nrml
+
+    :param request:
+        a `django.http.HttpRequest` object containing the mandatory
+        parameter 'xml_text': the text of the xml to be validated as nrml
+
+    :returns: a JSON object, containing:
+        * 'valid': a boolean indicating if the provided text is a valid nrml
+        * 'error_msg': the validation error message, if any error was found
+                       (None otherwise)
+        * 'error_line': line of the given xml where the error was found
+                        (None if no error was found)
+    """
+    xml_text = request.POST.get('xml_text')
+    if not xml_text:
+        return HttpResponseBadRequest(
+            'Please provide the "xml_text" parameter')
+    xml_file = writetmp(xml_text, suffix='.xml')
+    response_data = {}
+    try:
+        nrml.read(xml_file)
+    except Exception as exc:
+        error_msg = str(exc).split(', line')[0]
```

Comment:
Inside `exc` there could be an UTF8-encoded string, so `str(exc)` may fail. Do some experiments and find the right combination to manage that case.